accounts-db/src/in_mem_accounts_index.rs

```diff
@@ -672,14 +672,16 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
 
     /// Queue up these insertions for when the flush thread is dealing with this bin.
     /// This is very fast and requires no lookups or disk access.
-    pub fn startup_insert_only(&self, slot: Slot, items: impl Iterator<Item = (Pubkey, T)>) {
+    pub fn startup_insert_only(&self, items: impl Iterator<Item = (Pubkey, (Slot, T))>) {
         assert!(self.storage.get_startup());
         assert!(self.bucket.is_some());
 
         let mut insert = self.startup_info.insert.lock().unwrap();
+        // todo: memcpy the new slice into our vector already
+        // todo: avoid reallocs and just allocate another vec instead of likely resizing this one over and over
         items
             .into_iter()
-            .for_each(|(k, v)| insert.push((slot, k, v)));
+            .for_each(|(k, (slot, v))| insert.push((k, (slot, v))));
```

Comment:
in the existing code and new code we immediately. reformat to put the slot in there per element. slot pairs with `v`.